qark/modules/findPending.py

```diff
@@ -21,456 +22,499 @@
+                issue = ReportIssue()
+                issue.setCategory(ExploitType.INTENT)
+                issue.setDetails("PendingIntent created with implicit intent. File: " + str(current_file))
+                issue.setFile(current_file)
+                issue.setSeverity(Severity.VULNERABILITY)
+                results.append(issue)
+
+                issue = terminalPrint()
+                issue.setLevel(Severity.VULNERABILITY)
+                issue.setData("PendingIntent created with implicit intent. File: " + str(current_file))
+                results.append(issue)
+        else:
+            # TODO - Remove this (never seems to be hit)
+            report.write("parsingerror-issues-list", str(current_file), "strong")
+            common.logger.debug("Please report the following error, if you see this")
+            common.logger.debug(
+                "ERROR: INCOMPLETE CODE BRANCH REACHED findPending.py #4 - Press any key to continue (results may be incomplete)")
+    elif type(arg) is m.Name:
+        if hasattr(arg, 'value'):
+            try:
+                find_intent(arg.value, results)
+            except Exception as e:
+                report.write("parsingerror-issues-list", str(current_file), "strong")
+                common.logger.debug("Problem in find_intent function of findPending.py: " + str(e))
+    else:
+        # TODO - Remove this (never seems to be hit)
+        report.write("parsingerror-issues-list", str(current_file), "strong")
+        common.logger.debug("Please report the following error, if you see this")
+        common.logger.debug(
+            "ERROR: INCOMPLETE CODE BRANCH REACHED findPending.py #5 - Press any key to continue (results may be incomplete)")
+    return
+
+
+def find_class(classname, results):
+    found = False
+    """ Find the class name """
+    if hasattr(tree, 'type_declarations'):
+        for type_decl in tree.type_declarations:
+            if type(type_decl) is m.ClassDeclaration:
+                if str(type_decl.name) == str(classname):
+                    found = True
+                    return found
+                else:
+                    if recurse_class(classname, type_decl, results):
+                        return True
+            elif type(type_decl) is m.VariableDeclaration:
+                if hasattr(type_decl, 'type'):
+                    if hasattr(type_decl.type, 'name'):
+                        if hasattr(type_decl.type.name, 'value'):
+                            if str(type_decl.type.name.value) == str(classname):
+                                return True
+            else:
+                if recurse_class(classname, type_decl, results):
+                    return True
+    return found
+
+
+def recurse_class(classname, treeObject, results):
+    found = False
+    if hasattr(treeObject, '_fields'):
+        for x in treeObject._fields:
+            if type(getattr(treeObject, x)) is list:
```

Comment:
like above, this looks like the start of a switch statement where each function can be broken out into a seperate helper function